crates/bevy_math/src/bounding/bounded2d/mod.rs

```diff
@@ -1,6 +1,29 @@
+mod primitive_impls;
+
 use super::BoundingVolume;
 use crate::prelude::Vec2;
 
+/// Rotates the given vector counterclockwise by an angle in radians.
+#[inline(always)]
+pub(super) fn rotate_vec2(vec: Vec2, rotation: f32) -> Vec2 {
+    let (sin, cos) = rotation.sin_cos();
+    Vec2::new(vec.x * cos - vec.y * sin, vec.x * sin + vec.y * cos)
```

Comment:
Agreed. Although eventually we'll probably want a separate `Rotation2d` type or similar, since rotating vectors is *very* common in collision detection (and physics). A unit complex representation like what bevy_xpbd and (I think) Parry and Rapier use for 2D might be ideal in terms of memory and computation efficiency